http-client-core/src/main/java/io/micronaut/http/client/interceptor/HttpClientIntroductionAdvice.java

```diff
@@ -171,312 +174,421 @@ public Object intercept(MethodInvocationContext<Object, Object> context) {
+                if (argument.getAnnotationMetadata().hasStereotype(Format.class)) {
+                    conversionService.convert(value,
+                            ConversionContext.STRING.with(argument.getAnnotationMetadata()))
+                        .ifPresent(v -> pathParams.put(name, v));
+                } else {
+                    pathParams.put(name, value);
+                }
+            } else {
+                bodyArguments.add(ctx.getArgument());
+            }
+        };
+    }
 
-                        if (HttpResponse.class.isAssignableFrom(javaReturnType)) {
-                            return handleBlockingCall(javaReturnType, () ->
-                                    blockingHttpClient.exchange(request,
-                                            returnType.asArgument().getFirstTypeVariable().orElse(Argument.OBJECT_ARGUMENT),
-                                            errorType
-                                    ));
-                        } else if (void.class == javaReturnType) {
-                            return handleBlockingCall(javaReturnType, () ->
-                                    blockingHttpClient.exchange(request, null, errorType));
-                        } else {
-                            return handleBlockingCall(javaReturnType, () ->
-                                    blockingHttpClient.retrieve(request, returnType.asArgument(), errorType));
-                        }
-                    default:
-                        return interceptedMethod.unsupported();
+    @NonNull
+    private Optional<Object> bindArguments(MethodInvocationContext<Object, Object> context,
+                                           Map<String, MutableArgumentValue<?>> parameters,
+                                           ClientArgumentRequestBinder<Object> defaultBinder,
+                                           ClientRequestUriContext uriContext,
+                                           MutableHttpRequest<?> request,
+                                           InterceptedMethod interceptedMethod) {
+        Optional<Object> bindingErrorResult = Optional.empty();
+        Argument<?>[] arguments = context.getArguments();
+        for (Argument<?> argument : arguments) {
+            Object definedValue = getValue(argument, context, parameters);
+
+            if (definedValue != null) {
+                final ClientArgumentRequestBinder<Object> binder = (ClientArgumentRequestBinder<Object>) binderRegistry
+                    .findArgumentBinder((Argument<Object>) argument)
+                    .orElse(defaultBinder);
+                ArgumentConversionContext conversionContext = ConversionContext.of(argument);
+                binder.bind(conversionContext, uriContext, definedValue, request);
+                if (conversionContext.hasErrors()) {
+                    return Optional.of(interceptedMethod.handleException(new ConversionErrorException(argument, conversionContext.getLastError().get())));
```

Comment:
```suggestion
                    return conversionContext.getLastError().map(e -> interceptedMethod.handleException(new ConversionErrorException(argument, e)));
```

Does this appease Sonar (and do the same thing?)

Reply:
Yes, I think that's better. Thanks!